### ISSUE TITLE
Expose a `Producer<T>` generic type

### ISSUE THREAD

**Issue by ianstormtaylor** [immerjs/immer#955]
## 🚀 Feature Proposal

Right now it's really hard to figure out how to properly type a producer function. It would be really nice if there was an easy `Producer<T>` to refer to the producer function for a `Draft<T>`.

For example to pass a producer function around in React's context.

## Example

```ts
let ProducerContext = createContext<Producer<MyState> | null>(null)
```

**Comment by mweststrate** [comment]
Good idea! PR welcome :)

**Comment by ianstormtaylor** [comment]
@mweststrate I think there is a PR, although not sure the naming is quite right.

**Comment by NickSolante** [comment]
I saw there was an open PR for this but had a comment on it for rename and has since not been addressed can I just update that? what are the rules for this?

**Comment by tlylt** [comment]
This issue can probably be closed given that PR #968 has been merged
